feat: cross-platform backup - WPB-16658

caldrian opened the pull request:
<!--do not remove this marker, its needed to replace info when ticket title is updated -->
<!--jira-description-action-hidden-marker-start-->

<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-16658" title="WPB-16658" target="_blank"><img alt="Topic" src="https://wearezeta.atlassian.net/rest/api/2/universal_avatar/view/type/issuetype/avatar/15209?size=medium" />WPB-16658</a>  Cross Platform Backup
  </td></table>
  <br />
 

<!--jira-description-action-hidden-marker-end-->
<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove this marker, its needed to replace info when ticket title is updated -->

<!--do not remove the jira markers to link tickets automatically -->


### Issue

This PR:
- integrates a cross-platform framework for creating backups,
- adds a feature flag to create backups using the old way

### Testing

Create a backup and import it on other platforms clients.

---

### Checklist

- [ ] Title contains a reference JIRA issue number like `[WPB-XXX]`.
- [ ] Description is filled and free of optional paragraphs.
- [ ] Adds/updates automated tests.

---

### UI accessibility checklist

_If your PR includes UI changes, please utilize this checklist:_
- [ ] Make sure you use the API for UI elements that support large fonts.
- [ ] All colors are taken from WireDesign.ColorTheme or constructed using WireDesign.BaseColorPalette.
- [ ] New UI elements have Accessibility strings for VoiceOver.

caldrian committed:
Merge remote-tracking branch 'github/feat/cross-platform-backup-WPB-16658' into chore/export-cross-platform-backup-WPB-14592

caldrian committed:
add missing imports

caldrian committed:
fix data model

caldrian committed:
fix request strategy

caldrian committed:
fix domain

caldrian committed:
fix sync engine

caldrian committed:
fix app target

caldrian committed:
format code

caldrian committed:
Merge branch 'chore/unify-qualifiedid-WPB-14592' into chore/export-cross-platform-backup-WPB-14592

caldrian committed:
delete redundant mappings

caldrian committed:
minor fixes

caldrian committed:
Merge branch 'chore/unify-qualifiedid-WPB-14592' into chore/export-cross-platform-backup-WPB-14592

caldrian committed:
rename framework

caldrian committed:
fix build errors

caldrian committed:
update package versions

caldrian committed:
attempt to fix linking error

dmitrysimkin committed:
fix: Fix passing values for channel creation - WPB-17195 (#2930)

caldrian committed:
attempt to fix linking error

caldrian committed:
Merge branch 'release/cycle-3.123' into chore/unify-qualifiedid-WPB-14592

caldrian committed:
Merge branch 'chore/unify-qualifiedid-WPB-14592' into chore/export-cross-platform-backup-WPB-14592

caldrian committed:
minor cleanup/fixes

caldrian committed:
Trigger CI

caldrian committed:
create WireBackup package

caldrian committed:
add dependency

caldrian committed:
move framework

caldrian committed:
fix package configs

caldrian committed:
move some files

caldrian committed:
move protocols

caldrian committed:
create target kalium adapter

caldrian committed:
BackupCreator

caldrian committed:
delete kalium adapter target

caldrian committed:
migrate some more code

caldrian committed:
delete unused test

caldrian committed:
structural changes

caldrian committed:
minor improvements

caldrian committed:
undo project changes

caldrian committed:
fix linking errors

caldrian committed:
Merge branch 'chore/unify-qualifiedid-WPB-14592' into chore/export-cross-platform-backup-WPB-14592

jullianm committed:
chore: add debug action to enable/disable usage of CallKit for calling operations - WPB-17355 (#2938)

caldrian committed:
Merge branch 'release/cycle-3.123' into chore/unify-qualifiedid-WPB-14592

caldrian committed:
move files

caldrian committed:
move readme

caldrian committed:
delete placeholder test

caldrian committed:
clean up domain package files

caldrian committed:
move file

caldrian committed:
IndividualToTeamMigrationUseCaseAdapter

caldrian committed:
move files into WireUI

KaterinaWire committed:
feat: search button for 1:1 conversation - WPB-17294 (#2934)

Co-authored-by: Dmitry Simkin <dzmitry.simkin.ext@wire.com>

caldrian committed:
add adapters

caldrian committed:
delete WireDomain package

caldrian committed:
Merge branch 'release/cycle-3.123' of github.com:wireapp/wire-ios into chore/delete-wiredomainpackage-WPB-14598

caldrian committed:
minor fixes

caldrian committed:
rename type

caldrian committed:
fixes

caldrian committed:
format code

caldrian committed:
Merge branch 'chore/delete-wiredomainpackage-WPB-14598' into chore/unify-qualifiedid-WPB-14592

caldrian committed:
Trigger CI

caldrian committed:
undo linking

caldrian committed:
link WireFoundation

caldrian committed:
Merge branch 'chore/unify-qualifiedid-WPB-14592' into chore/export-cross-platform-backup-WPB-14592

# Conflicts:
#	WireBackup/Sources/WireBackup/Protocols/CreateBackup/CreateBackupUseCaseProtocol.swift
#	WireBackup/Sources/WireBackup/Protocols/Entities/Conversation/ConversationStoreProtocol.swift
#	WireBackup/Sources/WireBackup/Protocols/ImportBackup/ImportBackupProgress.swift
#	WireBackup/Sources/WireBackup/Protocols/ImportBackup/ImportBackupUseCaseProtocol.swift
#	WireBackup/Tests/WireBackupTests/WireBackupTests.swift
#	WireDomain/Package.swift
#	WireDomain/Sources/WireDomainPackage/Backup/Export/CreateLegacyBackupError.swift
#	WireDomain/Sources/WireDomainPackage/Backup/Import/ImportBackupError.swift
#	WireDomain/Sources/WireDomainPackage/UseCases/Protocols/CreateLegacyBackupError.swift
#	WireDomain/Sources/WireDomainPackage/UseCases/Protocols/ImportBackupError.swift
#	WireDomain/Sources/WireDomainPackage/UseCases/Protocols/ImportBackupProgress.swift
#	WireDomain/Sources/WireDomainPackage/UseCases/Protocols/ImportBackupUseCaseProtocol.swift
#	WireDomain/Tests/WireDomainPackageTests/PlaceholderTests.swift
#	WireUI/Sources/WireIndividualToTeamMigrationUI/UseCases/IndividualToTeamMigrationError.swift
#	WireUI/Sources/WireIndividualToTeamMigrationUI/UseCases/IndividualToTeamMigrationUseCaseProtocol.swift
#	WireUI/Sources/WireSettingsUI/Account/BackupImportExport/Protocols/CreateBackup/CreateBackupProgress.swift
#	WireUI/Sources/WireSettingsUI/Account/BackupImportExport/Protocols/CreateBackup/CreateBackupUseCaseProtocol.swift
#	WireUI/Sources/WireSettingsUI/Account/BackupImportExport/Protocols/CreateBackupProgress.swift
#	WireUI/Sources/WireSettingsUI/Account/BackupImportExport/Protocols/CreateBackupUseCaseProtocol.swift
#	WireUI/Sources/WireSettingsUI/Account/BackupImportExport/Protocols/ImportBackup/ImportBackupProgress.swift
#	WireUI/Sources/WireSettingsUI/Account/BackupImportExport/Protocols/ImportBackup/ImportBackupUseCaseProtocol.swift
#	WireUI/Sources/WireSettingsUI/Account/BackupImportExport/Protocols/ImportBackup/ImportLegacyBackupError.swift
#	wire-ios-sync-engine/Source/Use cases/ImportBackupUseCase/ImportBackupFileArchiver.swift
#	wire-ios-sync-engine/Source/Use cases/ImportBackupUseCase/SessionManager+importBackupUseCase.swift
#	wire-ios-sync-engine/Support/Sourcery/generated/AutoMockable.generated.swift
#	wire-ios-sync-engine/Tests/Source/Use cases/ImportBackupUseCaseTests.swift
#	wire-ios/Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift

caldrian committed:
Merge branch 'chore/delete-wiredomainpackage-WPB-14598' into chore/unify-qualifiedid-WPB-14592

caldrian committed:
chore: unify QualifiedID types - WPB-14592 (#2935)

caldrian committed:
Merge branch 'chore/unify-qualifiedid-WPB-14592.squashed' into chore/unify-qualifiedid-WPB-14592

caldrian committed:
Merge branch 'chore/unify-qualifiedid-WPB-14592' into chore/export-cross-platform-backup-WPB-14592

caldrian committed:
Revert "chore: unify QualifiedID types - WPB-14592 (#2935)"

This reverts commit 621ca7fa0a458d14c210b4a230aa0fc9e19b09ca.

caldrian committed:
fix build error

dmitrysimkin committed:
fix: Conversation header call buttons not re-set - WPB-15074 (#2937)

caldrian committed:
remove imports

caldrian committed:
fix WireUI tests

caldrian committed:
minor cleanup

caldrian committed:
Merge branch 'chore/delete-wiredomainpackage-WPB-14598' into chore/export-cross-platform-backup-WPB-14592

caldrian committed:
Merge branch 'release/cycle-3.123' into chore/delete-wiredomainpackage-WPB-14598

caldrian committed:
prepare for separating legacy and current backup import use case

caldrian committed:
apply new mock changes

caldrian committed:
Merge branch 'chore/update-sourcery-stencil-WPB-9200' into chore/export-cross-platform-backup-WPB-14592

caldrian committed:
attempt to restore legacy backup import

caldrian committed:
undo renaming project

caldrian committed:
restore WireDomain package target

caldrian committed:
format code

caldrian committed:
Merge branch 'chore/delete-wiredomainpackage-WPB-14598' into chore/export-cross-platform-backup-WPB-14592

# Conflicts:
#	WireDomain/Package.swift

caldrian committed:
add import

caldrian committed:
Trigger CI

caldrian committed:
chore: remove code from WireDomain package - WPB-14598 (#2945)

caldrian committed:
Merge branch 'chore/delete-wiredomainpackage-WPB-14598.squahsed' into chore/delete-wiredomainpackage-WPB-14598

caldrian committed:
clean up

caldrian committed:
Merge branch 'chore/delete-wiredomainpackage-WPB-14598' into chore/export-cross-platform-backup-WPB-14592

caldrian committed:
Revert "chore: remove code from WireDomain package - WPB-14598 (#2945)"

This reverts commit 824f9c29d4fdcf79802cda82271c680476f0f361.

# Conflicts:
#	WireUI/Tests/WireSettingsUITests/Account/BackupImportExport/Import/ImportBackupViewModelTest.swift
#	wire-ios-sync-engine/Source/Use cases/ImportBackupUseCase/ImportBackupFileArchiver.swift
#	wire-ios-sync-engine/Source/Use cases/ImportBackupUseCase/ImportLegacyBackupUseCase.swift
#	wire-ios-sync-engine/Source/Use cases/ImportBackupUseCase/SessionManager+importBackupUseCase.swift
#	wire-ios-sync-engine/Support/Sourcery/generated/AutoMockable.generated.swift
#	wire-ios/Wire-iOS/Sources/UserInterface/Settings/CellDescriptors/SettingsCellDescriptorFactory+Account.swift

caldrian committed:
link WireBackup

caldrian committed:
minor fixes

caldrian committed:
start over with CreateBackupUseCase

caldrian committed:
UserStoreProtocol+UserEntity

caldrian committed:
fix some build errors

caldrian committed:
count users and conversations

caldrian committed:
export conversations

jullianm committed:
fix: sync status bar never goes away on error - WPB-17151 (#2948)

caldrian committed:
some cleanup

caldrian committed:
minor fix

caldrian committed:
CreateBackupUseCaseAdapter

caldrian committed:
attempt to map user and conversation entities

caldrian committed:
export users and conversations

caldrian committed:
prepare for exporting messages

caldrian committed:
fix a bug

caldrian committed:
Merge branch 'release/cycle-3.123' into feat/cross-platform-backup-WPB-16658

caldrian committed:
Merge remote-tracking branch 'github/feat/cross-platform-backup-WPB-16658' into chore/export-cross-platform-backup-WPB-14592

caldrian committed:
Merge branch 'chore/export-cross-platform-backup-WPB-14592' into feat/cross-platform-backup-WPB-16658

caldrian committed:
Merge branch 'release/cycle-3.123' into chore/update-sourcery-stencil-WPB-9200

caldrian committed:
Merge branch 'chore/update-sourcery-stencil-WPB-9200' into feat/cross-platform-backup-WPB-16658

KaterinaWire committed:
feat: Connect with user via profile link or QR code - WPB-17342 (#2950)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Sam Wyndham <samwyndham@users.noreply.github.com>

caldrian committed:
minor cleanup

caldrian committed:
remove TODO

caldrian committed:
cleanup

caldrian committed:
check for cancellation

caldrian committed:
fix dependency graph

caldrian committed:
undo temporary changes

caldrian committed:
delete unused files

caldrian committed:
format code

caldrian committed:
more cleanup

caldrian committed:
upgrade swiftlint and swiftformat

caldrian committed:
set updated versions

caldrian committed:
format code

caldrian committed:
update docker images

caldrian committed:
downgrade

caldrian committed:
Merge branch 'chore/update-code-formatting-tools-WPB-17413' into feat/cross-platform-backup-WPB-16658

caldrian committed:
Merge branch 'chore/update-code-formatting-tools-WPB-17413' into chore/update-sourcery-stencil-WPB-9200

caldrian committed:
fix build errors

caldrian committed:
chore: use new AutoMockable stencil in WireFoundation - WPB-9200 (#2947)

caldrian committed:
Merge branch 'chore/update-sourcery-stencil-WPB-9200' into chore/update-code-formatting-tools-WPB-17413

caldrian committed:
Merge branch 'chore/update-code-formatting-tools-WPB-17413' into feat/cross-platform-backup-WPB-16658

caldrian committed:
cleanup

caldrian committed:
Trigger CI

caldrian committed:
create test file structure

caldrian committed:
Trigger CI

caldrian committed:
update framework

caldrian committed:
delete KMP-NativeCoroutines

caldrian committed:
add TODO

caldrian committed:
attempt to fix build error

caldrian committed:
chore: update code formatting tools - WPB-17413 (#2953)

caldrian committed:
Merge remote-tracking branch 'github/release/cycle-3.123' into chore/update-code-formatting-tools-WPB-17413

caldrian committed:
Merge branch 'chore/update-code-formatting-tools-WPB-17413' into feat/cross-platform-backup-WPB-16658

samwyndham reviewed (approved):
Nice work! Just left a few questions/suggestions

caldrian committed:
use percent format style

caldrian committed:
remove redundant properties and init

caldrian commented on WireBackup/Sources/WireBackup/Protocols/CreateBackup/CreateBackupProgress.swift:
```diff
@@ -0,0 +1,38 @@
+
+}
+
+public extension CreateBackupProgress {
+
+    static func progress(_ current: Int, _ total: Int) -> Self {
+        .progress(current: current, total: total)
+    }
+
+    static func done(_ url: URL) -> Self {
+        .done(url: url)
+    }
+
+}
```
They allow to put or leave the labels (current, total):
```
let p = CreateBackupProgress.progress(current: 1, total: 2)
let p = CreateBackupProgress.progress(1, 2)

#expect(p == (1, 2))
#expect(p == (current: 1, total: 2))
```

caldrian committed:
apply suggestion

caldrian commented on WireFoundation/Sources/WireFoundation/BaseTypes/QualifiedID.swift:
```diff
@@ -0,0 +1,63 @@
+            id: uuid,
+            domain: domain
+        )
+    }
```
done!

caldrian commented on WireBackup/Sources/WireBackup/KaliumBackupAdapters/BackupCreator/BackupCreator+addMessage.swift:
```diff
@@ -0,0 +1,121 @@
+
+    }
+
+    private func BackupMessageContent(_ content: MessageContent) -> KaliumBackup.BackupMessageContent {
```
done

caldrian committed:
fix displaying progress

caldrian committed:
update snapshot reference images

caldrian committed:
fix build errors

caldrian commented on WireDomain/Sources/WireDomain/Repositories/User/Protocols/UserRepositoryProtocol.swift:
```diff
@@ -150,4 +150,9 @@ public protocol UserRepositoryProtocol {
 
     func fetchAllUserIDsWithOneOnOneConversation() async throws -> [WireDataModel.QualifiedID]
 
+//    /// Fetches all user IDs from the local storage.
+//    /// - returns: A list of users' qualified IDs.
+//
+//    func fetchAllUserIDs() async throws -> [WireDataModel.QualifiedID]
```
yes, thanks!

caldrian commented on WireDomain/Sources/WireDomain/Repositories/Conversations/LocalStore/ConversationLocalStore.swift:
```diff
@@ -550,6 +556,12 @@ public final class ConversationLocalStore: ConversationLocalStoreProtocol {
         }
     }
 
+    public func fetchAllBackupableConversations() async throws -> [ZMConversation] {
```
done

caldrian committed:
apply changes from other branch

caldrian commented on WireBackup/Sources/WireBackup/UseCases/CreateBackupUseCase.swift:
```diff
@@ -0,0 +1,193 @@
+import Foundation
+@preconcurrency import KaliumBackup
+
+public struct CreateBackupUseCase<
+    UserStore: UserStoreProtocol,
+    ConversationStore: ConversationStoreProtocol,
+    MessageStore: MessageStoreProtocol,
+    FileArchiver: FileArchiverProtocol
+>: CreateBackupUseCaseProtocol {
+
```
Sure it could, but why do you prefer it without generics?
Generics come with some advantages (probably minimal but still).
SwiftUI makes heavy use of generics.

dmitrysimkin replied:
There using generic has a reason, is there a benefit of additional generic?

caldrian committed:
apply changes from other branch

caldrian committed:
apply changes from other branch

caldrian committed:
delete TODOs

caldrian committed:
fix build errors

caldrian committed:
delete TODOs

caldrian committed:
change developer flag

caldrian committed:
minor corrections

caldrian committed:
move CreateBackupUseCaseProtocol to WireFoundation

caldrian committed:
fix crash

caldrian committed:
rename flag

netbe commented:
do not merge until @caldrian comes back, not for current release

KaterinaWire committed:
fix: app sends large amount of last read messages - WPB-17439 (#2968)

Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: John Nguyen <28632506+johnxnguyen@users.noreply.github.com>

KaterinaWire committed:
feat: 1:1 proteus is not migrated to MLS - WPB-17393 (#2967)

KaterinaWire committed:
feat: QR code Profile link - Federation - update domain - WPB-10531 (#2970)

KaterinaWire committed:
feat: Update deep link format for QR code - WPB-10531 (#2978)

dmitrysimkin committed:
feat: Update max participants to 2000 - WPB-17344 (#2983)

johnxnguyen committed:
refactor: disable background fetch - WPB-15777 (#2985)

johnxnguyen committed:
fix: events received via push channel are refetched - WPB-17602 (#2987)

johnxnguyen committed:
fix: don't update last event id for transient live events - WPB-17602 (#2989)

netbe committed:
fix: channel icon - WPB-17556 (#2982)

johnxnguyen committed:
fix: event processor is missing some event consumers - WPB-17587 (#2994)

johnxnguyen committed:
fix: update event migration missed member removed events - WPB-17588 (#2995)

jullianm committed:
fix: channel deletion error pop up - WPB-17590 (#2997)

samwyndham committed:
chore: fix failing test on develop - WPB-17624 (#3000)

caldrian committed:
Merge branch 'release/cycle-3.123' into feat/cross-platform-backup-WPB-16658

caldrian committed:
feat: cross-platform backup - WPB-16658 (#2889)

# Conflicts:
#	WireDomain/Package.swift
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedColorSchemeVariants.dark.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.accessibility1.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.accessibility2.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.accessibility3.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.accessibility4.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.accessibility5.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.large.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.medium.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.small.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.xLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.xSmall.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.xxLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.xxxLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingColorSchemeVariants.dark.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.accessibility1.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.accessibility2.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.accessibility3.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.accessibility4.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.accessibility5.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.large.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.medium.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.small.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.xLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.xSmall.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.xxLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.xxxLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testColorSchemeVariants.dark.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.accessibility1.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.accessibility2.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.accessibility3.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.accessibility4.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.accessibility5.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.large.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.medium.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.small.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.xLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.xSmall.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.xxLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.xxxLarge.png
#	wire-ios/Wire-iOS.xcodeproj/project.pbxproj

caldrian committed:
feat: cross-platform backup - WPB-16658 (#2889)

caldrian committed:
Merge branch 'feat/cross-platform-backup-WPB-16658' into feat/cross-platform-backup-WPB-16658.bak-123

dmitrysimkin committed:
fix: channel icon Snapshot tests - WPB-17556 (#3001)

caldrian committed:
Merge commit '51c4df23f395466ac8ea1b22c69ce6095e69a994' into feat/cross-platform-backup-WPB-16658

# Conflicts:
#	WireDomain/Package.swift
#	WireDomain/Sources/WireDomain/Repositories/UpdateEvents/UpdateEventsLocalStore.swift
#	WireDomain/Sources/WireUpdateEventCoding/Utilities/Array+Utilities.swift
#	WireDomain/Tests/WireDomainTests/LocalStores/UpdateEventsLocalStoreTests.swift
#	WireFoundation/Sources/WireFoundation/Utilities/FileArchiver/FileArchivingError.swift
#	WireFoundation/Sources/WireFoundation/Utilities/FileArchiver/FileUnarchiverProtocol.swift
#	WireUI/Sources/WireSettingsUI/Account/BackupImportExport/Protocols/CreateBackupProgress.swift
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedColorSchemeVariants.dark.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.accessibility1.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.accessibility2.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.accessibility3.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.accessibility4.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.accessibility5.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.large.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.medium.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.small.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.xLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.xSmall.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.xxLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testFinishedDynamicTypeVariants.xxxLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingColorSchemeVariants.dark.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.accessibility1.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.accessibility2.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.accessibility3.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.accessibility4.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.accessibility5.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.large.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.medium.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.small.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.xLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.xSmall.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.xxLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/CreatingBackupProgressViewSnapshotTests/testOngoingDynamicTypeVariants.xxxLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testColorSchemeVariants.dark.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.accessibility1.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.accessibility2.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.accessibility3.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.accessibility4.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.accessibility5.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.large.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.medium.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.small.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.xLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.xSmall.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.xxLarge.png
#	WireUI/Tests/WireSettingsUITests/Resources/ReferenceImages/ImportProgressViewSnapshotTests/testDynamicTypeVariants.xxxLarge.png
#	wire-ios-data-model/Source/MLS/MLSSyncDelegate.swift
#	wire-ios-data-model/Tests/MLS/MLSServiceTests.swift
#	wire-ios-sync-engine/Source/Synchronization/SyncAgentDelegate.swift
#	wire-ios-sync-engine/Source/Use cases/ImportBackupUseCase/Protocols/ImportBackupFileArchiverProtocol.swift
#	wire-ios-sync-engine/Source/UserSession/ZMUserSession/ZMUserSession.swift
#	wire-ios/Wire-iOS Tests/ConversationViewControllerSnapshotTests.swift
#	wire-ios/Wire-iOS Tests/ReferenceImages/SettingsTableViewControllerSnapshotTests/testForAccountGroupWithDisabledEditing_Federated.1.png
#	wire-ios/Wire-iOS Tests/ReferenceImages/SettingsTableViewControllerSnapshotTests/testForAccountGroupWithDisabledEditing_NotFederated.1.png
#	wire-ios/Wire-iOS Tests/ReferenceImages/SettingsTableViewControllerSnapshotTests/testForAccountGroup_Federated.1.png
#	wire-ios/Wire-iOS Tests/ReferenceImages/SettingsTableViewControllerSnapshotTests/testForAccountGroup_NotFederated.1.png
#	wire-ios/Wire-iOS.xcodeproj/project.pbxproj

caldrian committed:
Trigger CI

jullianm committed:
fix: channel icon issue - WPB-17589 (#2991)

dmitrysimkin committed:
fix: max participants 2000 for channel and 500 for group - WPB-17344 (#2996)

caldrian committed:
Merge branch 'release/cycle-3.124' into feat/cross-platform-backup-WPB-16658

caldrian committed:
feat: cross-platform backup - WPB-16658 (#2889)

caldrian committed:
feat: cross-platform backup - WPB-16658 (#2889)

caldrian committed:
Merge commit 'e9ae4fbe7773cc4936473d76aa87007d1399dc58' into feat/cross-platform-backup-WPB-16658

caldrian committed:
Merge commit 'a2336dc71b2d56878eba1814e9d7114b81e1330f' into feat/cross-platform-backup-WPB-16658

# Conflicts:
#	WireDomain/Sources/WireDomain/Synchronization/IncrementalSync.swift
#	wire-ios/Wire-iOS/Sources/Helpers/syncengine/Conversation+Participants.swift

caldrian committed:
Merge branch 'develop' into feat/cross-platform-backup-WPB-16658

caldrian committed:
Trigger CI

caldrian committed:
minor refactoring

caldrian committed:
update framework

caldrian committed:
update reference

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/cross-platform-backup-WPB-16658

caldrian committed:
update framework

caldrian committed:
fix checksum

caldrian committed:
adjustments for the API changes

caldrian committed:
fix passing self user id

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/cross-platform-backup-WPB-16658

caldrian committed:
format code

caldrian committed:
fix build errors

caldrian committed:
update snapshot reference images

caldrian committed:
include new zip library

caldrian committed:
cleanup

caldrian committed:
add TODO

caldrian committed:
rename type

caldrian committed:
fix storing lowercased ids

caldrian committed:
attempt to fix crash

caldrian committed:
Merge branch 'develop' into feat/cross-platform-backup-WPB-16658

caldrian committed:
Merge remote-tracking branch 'github/feat/cross-platform-backup-WPB-16658' into feat/cross-platform-backup-WPB-16658

caldrian committed:
fix error, format code

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/cross-platform-backup-WPB-16658

# Conflicts:
#	wire-ios/Wire-iOS.xcodeproj/project.pbxproj

caldrian committed:
fix binary target reference

caldrian committed:
delete binary

caldrian committed:
chore: create package WireBackup - WPB-16658

caldrian committed:
Merge branch 'chore/create-wire-backup-package-WPB-16658' into chore/reference-kalium-xcframework-from-releases-WPB-16658

# Conflicts:
#	WireDomain/WireDomain Project.xcodeproj/project.pbxproj
#	wire-ios/Wire-iOS/Sources/UserInterface/Settings/Backup/QualifiedIDAdapters.swift

caldrian committed:
minor adjustments

caldrian committed:
Merge branch 'feat/cross-platform-backup-WPB-16658' into chore/reference-kalium-xcframework-from-releases-WPB-16658

caldrian committed:
update reference

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into chore/create-wire-backup-package-WPB-16658

caldrian committed:
add dependency to graph

caldrian committed:
more explicit test name

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into chore/create-wire-backup-package-WPB-16658

caldrian committed:
Merge branch 'chore/create-wire-backup-package-WPB-16658' into chore/reference-kalium-xcframework-from-releases-WPB-16658

caldrian committed:
Merge commit '46d92fbd1aca19f5c3da3d166047fd2f6560a913' into feat/cross-platform-backup-WPB-16658

caldrian committed:
update packages

johnxnguyen commented on WireBackup/Sources/WireBackup/UseCases/CreateBackupUseCase.swift:
```diff
@@ -0,0 +1,173 @@
+
+                    // get the counts of users, messages and conversations in order to report progress accurately
+                    logger.debug("calculating entity counts")
+                    let userCount = try await userStore.totalUserCount()
+                    let conversationCount = try await conversationStore.totalConversationCount()
+                    let messageCount = try await messageStore.totalMessageCount()
```
suggestion: I think having a single specific store that exposes the api necessary for the current task at hand may give the benefit of fewer context switches. For instance:

```swift
protocol BackupLocalStore {

    func countBackupModels() async throws -> (users: Int, conversations: Int, messages: Int)
    func fetchAllUsers() async throws -> Set<UserBackupModel>
    func fetchAllConversations() async throws -> Set<ConversationBackupModel>
    func fetchAllMessages() async throws -> Set<MessageBackupModel>
    ...

}
```

caldrian replied:
Done, thanks.

johnxnguyen commented on WireBackup/Sources/WireBackup/UseCases/CreateBackupUseCase.swift:
```diff
@@ -0,0 +1,193 @@
+                    try Task.checkCancellation()
+
+                    let messageProgressOffset = userCount + conversationCount
+                    let allMessages = try await messageStore.fetchAllMessages()
+                    for messageIndex in 0 ..< allMessages.count {
+                        backupCreator.addMessage(allMessages[messageIndex])
+                        if messageIndex % 50 == 0 { try Task.checkCancellation() }
+                        reportProgress(messageProgressOffset + messageIndex + 1, total)
+                    }
```
Yes, it may be worth to make `fetchAllMessages` operate in batches. Though I'm not sure how much of a difference it would make if the backupCreator is not being memory conscious.

johnxnguyen commented on WireDomain/Sources/WireDomain/Repositories/Conversations/LocalStore/ConversationLocalStore.swift:
```diff
@@ -1236,6 +1236,37 @@ public final class ConversationLocalStore: ConversationLocalStoreProtocol {
 
+    // MARK: - Backup / Restore
+
+    public func totalConversationCountForBackup() async throws -> Int {
+        try await context.perform { [context] in
+            try context.count(for: ZMConversation.fetchRequestForBackup())
+        }
+    }
+
+    public func fetchAllConversationIDsForBackup() async throws -> [QualifiedID] {
+        let fetchRequest = ZMConversation.fetchRequestForBackup()
+        fetchRequest.propertiesToFetch = ["remoteIdentifier_data", "domain"]
+        return try await context.perform { [context] in
+            let conversations = try context.fetch(fetchRequest)
+            return conversations.compactMap(\.qualifiedID)
+        }
+    }
+
+    public func fetchAllConversationsForBackup() async throws -> [ZMConversation] {
+        try await context.perform { [context] in
+            try context.fetch(ZMConversation.fetchRequestForBackup())
+        }
+    }
```
suggestion: as I have mentioned in a previous comment, I think it may be worth to have a specific backup store. I've discussed in the past how these entity based stores are growing and growing based on different needs and it's getting a bit messy. See for instance https://github.com/wireapp/wire-ios/blob/develop/wire-ios-sync-engine/Source/SessionManager/UpdateEventMigratorDAO.swift

caldrian replied:
Done, created a BackupLocalStoreProtocol as you suggested.

johnxnguyen commented on wire-ios-sync-engine/Source/Synchronization/EventProcessor.swift:
```diff
@@ -161,7 +161,6 @@ actor EventProcessor: UpdateEventProcessor {
     }
 
     private func enqueueTask(_ block: @escaping @Sendable () async throws -> Void) async throws {
-        defer { processingTask = nil }
```
question: why is this needed?

caldrian replied:
This statement could lead to the first execution clearing the task value a second execution has already set, in my opinion.

johnxnguyen commented on wire-ios/Wire-iOS/Sources/Helpers/ZipArchiving/ZIPFoundationFileArchiver.swift:
```diff
@@ -0,0 +1,63 @@
+            try? fileManager.removeItem(at: sourceURL)
+        }
+
+        // TODO: throwing an error here leads to the warning printed:
```
question: do we need to create a jira ticket for this?

johnxnguyen reviewed (commented):
Looking good, but I left some comments before approving.

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/cross-platform-backup-WPB-16658

caldrian committed:
fix warning

caldrian committed:
BackupLocalStoreProtocol

caldrian committed:
add deprecation

caldrian committed:
replace old protocols

caldrian committed:
fix build errors

caldrian committed:
migrate UserStoreAdapter

caldrian committed:
delete dead code

caldrian committed:
format code

caldrian committed:
delete comment

caldrian committed:
minor improvements

caldrian committed:
clean up

caldrian commented on WireBackup/Sources/WireBackup/UseCases/CreateBackupUseCase.swift:
```diff
@@ -0,0 +1,193 @@
+        self.logger = logger
+    }
+
+    public func invoke(password: String) -> AsyncThrowingStream<CreateBackupProgress, any Error> {
```
It's not easy because many types aren't sendable, so the compiler would quickly complain when extracting code.

caldrian committed:
clean up typealiases

caldrian committed:
include generic metadata

caldrian committed:
prepare for using AsyncThrowingStream for fetching entities

caldrian committed:
batch send

caldrian committed:
clean up

caldrian committed:
fix concurrency issue

caldrian committed:
batch conversation and message processing

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/cross-platform-backup-WPB-16658

caldrian committed:
disable new backup

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/cross-platform-backup-WPB-16658

johnxnguyen commented on WireDomain/Sources/WireDomain/Repositories/Conversations/LocalStore/ConversationLocalStore.swift:
```diff
@@ -1236,6 +1236,41 @@ public final class ConversationLocalStore: ConversationLocalStoreProtocol {
 
+    // MARK: - Backup / Restore
+
+    public func totalConversationCountForBackup() async throws -> Int {
```
Since you have a new specific store for backups, can we revert these changes?

caldrian replied:
done

johnxnguyen commented on WireDomain/Sources/WireDomain/Repositories/Message/MessageLocalStore.swift:
```diff
@@ -1021,4 +1021,29 @@ public final class MessageLocalStore: MessageLocalStoreProtocol {
 
+    // MARK: - Backup / Restore
+
+    public func totalMessageCountForBackup() async throws -> Int {
```
Since you have a new specific store for backups, can we revert these changes?

caldrian replied:
done

johnxnguyen commented on WireDomain/Sources/WireDomain/Repositories/User/UserLocalStore.swift:
```diff
@@ -413,4 +413,33 @@ public final class UserLocalStore: UserLocalStoreProtocol {
+
+    // MARK: - Backup / Restore
+
+    public func totalUserCountForBackup() async throws -> Int {
```
Since you have a new specific store for backups, can we revert these changes?

johnxnguyen commented on wire-ios/Wire-iOS/Sources/UserInterface/Settings/Backup/BackupLocalStore/BackupLocalStore.swift:
```diff
@@ -0,0 +1,170 @@
+    private let messageLocalStore: MessageLocalStore
+
+    func countModels() async throws -> (userCount: Int, conversationCount: Int, messageCount: Int) {
+        let userCount = try await userLocalStore.totalUserCountForBackup()
+        let conversationCount = try await conversationLocalStore.totalConversationCountForBackup()
+        let messageCount = try await messageLocalStore.totalMessageCountForBackup()
```
suggestion: rather than wrapping these stores, I suggest to take their implementations (which were added for the backup feature) and have them here in a single `await context.perform { ... }` block so we don't switch back and forth unnecessarily.

caldrian replied:
done 2ff3ec2c48e494db7de5418f00d22f4e43dd8447

johnxnguyen commented on wire-ios/Wire-iOS/Sources/UserInterface/Settings/Backup/BackupLocalStore/BackupLocalStore.swift:
```diff
@@ -0,0 +1,170 @@
+        AsyncThrowingStream { continuation in
+            Task<Void, Never> {
+                do {
+                    let messages = try await messageLocalStore.fetchAllMessagesForBackup()
```
issue: the fetch request inside this method has a batch limit, so `messages` is actually a batched array. We then take this array and iterate through all elements and push them through the stream. I'm wondering if by doing this we eliminate the benefits of the batching since those elements may all accumulate in memory. What do you think?

johnxnguyen replied:
The batching happens under the hood: it looks like an array of all fetched messages, but in reality only the first batch is initially loaded and the next batch is loaded lazily when that array index is accessed. Certainly we don't load all elements at once, but I guess we do load them all in to memory as we go through the array, I'm just wondering if there's some clean up we need to do to ensure the memory consumption reasonable.

In any case, it would be good to try to profile it to get an idea of what's going on. Also, in case we do get a memory warning, what would happen?

caldrian committed:
don't wrap stores

caldrian committed:
Merge branch 'develop' of github.com:wireapp/wire-ios into feat/cross-platform-backup-WPB-16658

johnxnguyen commented on WireDomain/Sources/WireDomain/Repositories/Conversations/Protocols/ConversationLocalStoreProtocol.swift:
```diff
@@ -16,6 +16,8 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
+public import CoreData
```
I guess this is not needed any more.

johnxnguyen commented on WireDomain/Sources/WireDomain/Repositories/Message/Protocols/MessageLocalStoreProtocol.swift:
```diff
@@ -16,6 +16,8 @@
 // along with this program. If not, see http://www.gnu.org/licenses/.
 //
 
+public import CoreData
```
I guess this is not needed any more.

dmitrysimkin reviewed (approved):
LGTM

johnxnguyen reviewed (approved):
Looks good! Just left some more comments

caldrian committed:
clean up

caldrian committed:
feat: cross-platform backup - WPB-16658 (#2889)

Co-authored-by: Jullian Mercier <31648126+jullianm@users.noreply.github.com>
Co-authored-by: Dmitry Simkin <dzmitry.simkin.ext@wire.com>
Co-authored-by: KaterinaWire <57407805+KaterinaWire@users.noreply.github.com>
Co-authored-by: github-actions[bot] <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Sam Wyndham <samwyndham@users.noreply.github.com>
Co-authored-by: John Nguyen <28632506+johnxnguyen@users.noreply.github.com>
Co-authored-by: François Benaiteau <netbe@users.noreply.github.com>

caldrian committed:
Merge remote-tracking branch 'github/gh-readonly-queue/develop/pr-2889-aec5b82f5d5da01bf823136099fde161c73c0b52' into feat/cross-platform-backup-WPB-16658

caldrian committed:
Merge branch 'release/cycle-3.125' of github.com:wireapp/wire-ios into feat/cross-platform-backup-WPB-16658

caldrian committed:
Merge branch 'release/cycle-3.125' of github.com:wireapp/wire-ios into feat/cross-platform-backup-WPB-16658
